addons/sale/models/sale.py

```diff
@@ -290,9 +290,8 @@ def _compute_expected_date(self):
         """
         for order in self:
             dates_list = []
-            confirm_date = fields.Datetime.from_string(order.date_order if order.state in ['sale', 'done'] else fields.Datetime.now())
```

Comment:
@grindtildeath Not to change too much and to avoid recomputing confirm_date per line, add confirm_date as parameter to the 'expected_date()' function.